tests/test_petsc.py

```diff
@@ -362,3 +362,66 @@ def test_separate_eqn():
         ' + Derivative(f2(t, x, y), (y, 2)) + f2(t, x, y)/dt'
     assert str(F4) == '-Derivative(f2(t + dt, x, y), (x, 2)) -' + \
         ' Derivative(f2(t + dt, x, y), (y, 2)) + f2(t + dt, x, y)/dt'
+
+
+@skipif('petsc')
+def test_centre_stencil():
```

Comment:
This should be done with `pytest.mark.parametrize` rather than hardcoding a bunch of values in a single test

Reply:
Also need to test something like `(f * g.dx).dy` to mimic div(grad) eqs.